src/vaadin-dialog.html

```diff
@@ -289,6 +391,172 @@
+                  if (width > minimumSize) {
+                    this._setBounds({width});
+                  }
+                  break;
+                }
+                case 's': {
+                  const height = originalBounds.height + (e.pageY - originalMouse.top);
+                  if (height > minimumSize) {
+                    this._setBounds({height});
+                  }
+                  break;
+                }
+                case 'w': {
+                  const width = originalBounds.width - (e.pageX - originalMouse.left);
+                  const left = originalBounds.left + (e.pageX - originalMouse.left);
+                  if (width > minimumSize) {
+                    this._setBounds({left, width});
+                  }
+                  break;
+                }
+              }
+            });
+          };
+          const stopResize = edge => window.removeEventListener('mousemove', resizeEvents[edge]);
+          const resizeEvents = {};
+          const stopResizeEvents = {};
+
+          // Note: edge controls added before corners
+          ['n', 'e', 's', 'w', 'nw', 'ne', 'se', 'sw'].forEach(edge => {
+            const resizer = document.createElement('div');
+            if (edge.length === 1) {
+              resizer.classList.add('edge');
+            }
+            resizer.classList.add('resizer');
+            resizer.classList.add(edge);
+            resizer.addEventListener('mousedown', e => {
+              e.preventDefault();
+              e.stopPropagation();
+              this._moveOverlayToTop();
+              const overlayBounds = overlay.getBoundingClientRect();
+              const containerBounds = this.$.overlay.getBoundingClientRect();
+              originalBounds.top = overlayBounds.top - containerBounds.top;
+              originalBounds.left = overlayBounds.left - containerBounds.left;
+              originalBounds.width = overlayBounds.width;
+              originalBounds.height = overlayBounds.height;
+              originalMouse.top = e.pageY;
+              originalMouse.left = e.pageX;
+              resizeEvents[edge] = e => resize(e, edge);
+              stopResizeEvents[edge] = () => stopResize(edge);
+              window.addEventListener('mousemove', resizeEvents[edge]);
+              window.addEventListener('mouseup', stopResizeEvents[edge]);
+            });
+            overlay.appendChild(resizer);
+          });
+        }
+
+        _setBounds({top, right, bottom, left, width, height}) {
+          const overlay = this.$.overlay.$.overlay;
+          const bounds = {top, right, bottom, left, width, height};
+
+          if (overlay.style.position !== 'absolute') {
+            overlay.style.position = 'absolute';
+          }
+
+          for (const arg in bounds) {
+            if (typeof bounds[arg] === 'number') {
+              bounds[arg] = `${bounds[arg]}px`;
+            }
+          }
+
+          if (bounds.top) {
```

Comment:
These lines could maybe be replaced with `Object.assign(overlay.style, bounds);`

Reply:
Done.